Fail
file path is :
projects/deltachat-core/rust/src/chat.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error: expected identifier, found `:`
    --> src/chat.rs:3750:8
     |
3750 | cleanup:
     |        ^ expected identifier

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3694:9
     |
3694 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3702:9
     |
3702 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3707:9
     |
3707 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3717:9
     |
3717 |         goto_cleanup!();
     |         ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3723:13
     |
3723 |             goto_cleanup!();
     |             ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3728:13
     |
3728 |             goto_cleanup!();
     |             ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3731:13
     |
3731 |             goto_cleanup!();
     |             ^^^^^^^^^^^^

error: cannot find macro `goto_cleanup` in this scope
    --> src/chat.rs:3734:13
     |
3734 |             goto_cleanup!();
     |             ^^^^^^^^^^^^

error[E0425]: cannot find value `DC_CONTEXT_MAGIC` in this scope
    --> src/chat.rs:3693:31
     |
3693 |     if context.inner.magic != DC_CONTEXT_MAGIC || contact.is_none() || chat_id <= DC_CHAT_ID_LAST_SPECIAL {
     |                               ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_ERROR_SELF_NOT_IN_GROUP` in this scope
    --> src/chat.rs:3706:28
     |
3706 |         context.emit_event(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Cannot add contact to group; self not in group.").await;
     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_UNPROMOTED` in this scope
    --> src/chat.rs:3710:45
     |
3710 |     if from_handshake && chat.param.get_int(DC_PARAM_UNPROMOTED, 0) == 1 {
     |                                             ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_UNPROMOTED` in this scope
    --> src/chat.rs:3711:24
     |
3711 |         chat.param.set(DC_PARAM_UNPROMOTED, None);
     |                        ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_BIDIRECT_VERIFIED` in this scope
    --> src/chat.rs:3726:74
     |
3726 |         if chat.is_protected() && contact.is_verified(context).await? != DC_BIDIRECT_VERIFIED {
     |                                                                          ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_ERROR_SELF_NOT_IN_GROUP` in this scope
    --> src/chat.rs:3727:29
     |
3727 | ...   context.log_err(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Only bidirectional verified contacts can be added to verified groups.").await;
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_STR_MSGADDMEMBER` in this scope
    --> src/chat.rs:3740:49
     |
3740 |         msg.text = dc_stock_system_msg(context, DC_STR_MSGADDMEMBER, contact.addr.as_str(), None, ContactId::SELF).await;
     |                                                 ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_CMD` in this scope
    --> src/chat.rs:3741:27
     |
3741 |         msg.param.set_int(DC_PARAM_CMD, DC_CMD_MEMBER_ADDED_TO_GROUP);
     |                           ^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CMD_MEMBER_ADDED_TO_GROUP` in this scope
    --> src/chat.rs:3741:41
     |
3741 |         msg.param.set_int(DC_PARAM_CMD, DC_CMD_MEMBER_ADDED_TO_GROUP);
     |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_CMD_ARG` in this scope
    --> src/chat.rs:3742:23
     |
3742 |         msg.param.set(DC_PARAM_CMD_ARG, contact.addr.as_str());
     |                       ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_PARAM_CMD_ARG2` in this scope
    --> src/chat.rs:3743:27
     |
3743 |         msg.param.set_int(DC_PARAM_CMD_ARG2, flags);
     |                           ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `flags` in this scope
    --> src/chat.rs:3743:46
     |
3743 |         msg.param.set_int(DC_PARAM_CMD_ARG2, flags);
     |                                              ^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_CHAT_MODIFIED` in this scope
    --> src/chat.rs:3746:24
     |
3746 |     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |                        ^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no function or associated item named `new_untyped` found for struct `message::Message` in the current scope
    --> src/chat.rs:3690:28
     |
3690 |     let mut msg = Message::new_untyped(context);
     |                            ^^^^^^^^^^^ function or associated item not found in `Message`
     |
    ::: src/message.rs:420:1
     |
420  | pub struct Message {
     | ------------------ function or associated item `new_untyped` not found for this struct
     |
note: if you're trying to build a new `message::Message`, consider using `message::Message::new` which returns `message::Message`
    --> src/message.rs:469:5
     |
469  |     pub fn new(viewtype: Viewtype) -> Self {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0609]: no field `magic` on type `Arc<InnerContext>`
    --> src/chat.rs:3693:22
     |
3693 |     if context.inner.magic != DC_CONTEXT_MAGIC || contact.is_none() || chat_id <= DC_CHAT_ID_LAST_SPECIAL {
     |                      ^^^^^ unknown field
     |
     = note: available fields are: `blobdir`, `sql`, `smeared_timestamp`, `generating_key_mutex`, `oauth2_mutex` ... and 17 others

error[E0599]: no method named `is_none` found for struct `Contact` in the current scope
    --> src/chat.rs:3693:59
     |
3693 |     if context.inner.magic != DC_CONTEXT_MAGIC || contact.is_none() || chat_id <= DC_CHAT_ID_LAST_SPECIAL {
     |                                                           ^^^^^^^ method not found in `Contact`
     |
    ::: src/contact.rs:358:1
     |
358  | pub struct Contact {
     | ------------------ method `is_none` not found for this struct

error[E0599]: no method named `reset_gossiped_timestamp` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3697:13
     |
3697 |     context.reset_gossiped_timestamp(chat_id).await?;
     |             ^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&Context`

error[E0425]: cannot find function `real_group_exists` in this scope
    --> src/chat.rs:3699:8
     |
3699 |     if real_group_exists(context, chat_id).await? == 0
     |        ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `real_contact_exists_by_id` in this scope
    --> src/chat.rs:3700:13
     |
3700 |         || (real_contact_exists_by_id(context, contact_id).await? == 0 && contact_id != ContactId::SELF)
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `load_from_db` found for struct `Chat` in the current scope
    --> src/chat.rs:3701:17
     |
1480 | pub struct Chat {
     | --------------- method `load_from_db` not found for this struct
...
3701 |         || chat.load_from_db(context).await? == 0 {
     |            -----^^^^^^^^^^^^---------
     |            |    |
     |            |    this is an associated function, not a method
     |            help: use associated function syntax instead: `Chat::load_from_db(context)`
     |
     = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
note: the candidate is defined in an impl for the type `Chat`
    --> src/chat.rs:1515:5
     |
1515 |     pub async fn load_from_db(context: &Context, chat_id: ChatId) -> Result<Self> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0061]: this method takes 1 argument but 3 arguments were supplied
    --> src/chat.rs:3706:17
     |
3706 |         context.emit_event(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Cannot add contact to group; self not in group.").await;
     |                 ^^^^^^^^^^                                   -  ------------------------------------------------- unexpected argument of type `&'static str`
     |                                                              |
     |                                                              unexpected argument of type `{integer}`
     |
note: method defined here
    --> src/context.rs:595:12
     |
595  |     pub fn emit_event(&self, event: EventType) {
     |            ^^^^^^^^^^        ----------------
help: remove the extra arguments
     |
3706 -         context.emit_event(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Cannot add contact to group; self not in group.").await;
3706 +         context.emit_event(DC_EVENT_ERROR_SELF_NOT_IN_GROUP).await;
     |

error[E0277]: `()` is not a future
    --> src/chat.rs:3706:116
     |
3706 |         context.emit_event(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Cannot add contact to group; self not in group.").await;
     |                                                                                                                   -^^^^^
     |                                                                                                                   ||
     |                                                                                                                   |`()` is not a future
     |                                                                                                                   help: remove the `.await`
     |
     = help: the trait `futures::Future` is not implemented for `()`, which is required by `(): std::future::IntoFuture`
     = note: () must be a future or must implement `IntoFuture` to be awaited
     = note: required for `()` to implement `std::future::IntoFuture`

error[E0061]: this method takes 1 argument but 2 arguments were supplied
    --> src/chat.rs:3710:37
     |
3710 |     if from_handshake && chat.param.get_int(DC_PARAM_UNPROMOTED, 0) == 1 {
     |                                     ^^^^^^^                    ---
     |                                                                | |
     |                                                                | unexpected argument of type `{integer}`
     |                                                                help: remove the extra argument
     |
note: method defined here
    --> src/param.rs:319:12
     |
319  |     pub fn get_int(&self, key: Param) -> Option<i32> {
     |            ^^^^^^^        ----------

error[E0308]: mismatched types
    --> src/chat.rs:3710:72
     |
3710 |     if from_handshake && chat.param.get_int(DC_PARAM_UNPROMOTED, 0) == 1 {
     |                          ------------------------------------------    ^ expected `Option<i32>`, found integer
     |                          |
     |                          expected because this is `std::option::Option<i32>`
     |
     = note: expected enum `std::option::Option<i32>`
                found type `{integer}`
help: try wrapping the expression in `Some`
     |
3710 |     if from_handshake && chat.param.get_int(DC_PARAM_UNPROMOTED, 0) == Some(1) {
     |                                                                        +++++ +

error[E0277]: `std::option::Option<_>` doesn't implement `std::fmt::Display`
    --> src/chat.rs:3711:45
     |
3711 |         chat.param.set(DC_PARAM_UNPROMOTED, None);
     |                    ---                      ^^^^ `std::option::Option<_>` cannot be formatted with the default formatter
     |                    |
     |                    required by a bound introduced by this call
     |
     = help: the trait `std::fmt::Display` is not implemented for `std::option::Option<_>`, which is required by `std::option::Option<_>: ToString`
     = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
     = note: required for `std::option::Option<_>` to implement `ToString`
note: required by a bound in `param::Params::set`
    --> src/param.rs:287:51
     |
287  |     pub fn set(&mut self, key: Param, value: impl ToString) -> &mut Self {
     |                                                   ^^^^^^^^ required by this bound in `Params::set`

error[E0599]: no method named `get_config` found for struct `Sql` in the current scope
    --> src/chat.rs:3715:34
     |
3715 |     self_addr = Some(context.sql.get_config("configured_addr", "").to_string());
     |                                  ^^^^^^^^^^ help: there is a method with a similar name: `get_raw_config`
     |
    ::: src/sql.rs:57:1
     |
57   | pub struct Sql {
     | -------------- method `get_config` not found for this struct

error[E0616]: field `addr` of struct `Contact` is private
    --> src/chat.rs:3716:16
     |
3716 |     if contact.addr.eq_ignore_ascii_case(&self_addr.unwrap()) {
     |                ^^^^ private field

error[E0599]: no method named `is_contact_in_chat` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3720:16
     |
3720 |     if context.is_contact_in_chat(chat_id, contact_id).await? {
     |                ^^^^^^^^^^^^^^^^^^ method not found in `&Context`

error[E0599]: no method named `log_err` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3727:21
     |
3727 | ...   context.log_err(DC_EVENT_ERROR_SELF_NOT_IN_GROUP, 0, "Only bidirectional verified contacts can be added to verified groups.").await;
     |               ^^^^^^^ method not found in `&Context`
     |
     = help: items from traits can only be used if the trait is implemented and in scope
note: `LogExt` defines an item `log_err`, perhaps you need to implement it
    --> src/log.rs:64:1
     |
64   | pub trait LogExt<T, E>
     | ^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `is_contact_in_chat` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3730:20
     |
3730 |         if context.is_contact_in_chat(chat_id, contact_id).await? {
     |                    ^^^^^^^^^^^^^^^^^^ method not found in `&Context`

error[E0599]: no method named `add_to_chat_contacts_table` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3733:20
     |
3733 |         if context.add_to_chat_contacts_table(chat_id, &[contact_id]).await? == 0 {
     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&Context`

error[E0277]: can't compare `bool` with `{integer}`
    --> src/chat.rs:3738:60
     |
3738 |     if chat.typ == Chattype::Group && chat.is_unpromoted() == 0 {
     |                                                            ^^ no implementation for `bool == {integer}`
     |
     = help: the trait `PartialEq<{integer}>` is not implemented for `bool`
     = help: the following other types implement trait `PartialEq<Rhs>`:
               <bool as PartialEq>
               <bool as PartialEq<bitvec::ptr::proxy::BitRef<'_, M, T, O>>>
               <bool as PartialEq<serde_json::Value>>
               <&bool as PartialEq<bitvec::ptr::proxy::BitRef<'_, M, T, O>>>

error[E0308]: mismatched types
    --> src/chat.rs:3738:63
     |
3738 |     if chat.typ == Chattype::Group && chat.is_unpromoted() == 0 {
     |                                                               ^ expected `bool`, found integer

error[E0616]: field `addr` of struct `Contact` is private
    --> src/chat.rs:3740:78
     |
3740 |         msg.text = dc_stock_system_msg(context, DC_STR_MSGADDMEMBER, contact.addr.as_str(), None, ContactId::SELF).await;
     |                                                                              ^^^^ private field

error[E0425]: cannot find function `dc_stock_system_msg` in this scope
    --> src/chat.rs:3740:20
     |
3740 |         msg.text = dc_stock_system_msg(context, DC_STR_MSGADDMEMBER, contact.addr.as_str(), None, ContactId::SELF).await;
     |                    ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0616]: field `addr` of struct `Contact` is private
    --> src/chat.rs:3742:49
     |
3742 |         msg.param.set(DC_PARAM_CMD_ARG, contact.addr.as_str());
     |                                                 ^^^^ private field

error[E0599]: no method named `is_none` found for struct `contact::Contact` in the current scope
    --> src/chat.rs:3693:59
     |
3693 |     if context.inner.magic != DC_CONTEXT_MAGIC || contact.is_none() || chat_id <= DC_CHAT_ID_LAST_SPECIAL {
     |                                                           ^^^^^^^ method not found in `Contact`
     |
    ::: src/contact.rs:358:1
     |
358  | pub struct Contact {
     | ------------------ method `is_none` not found for this struct

error[E0599]: no method named `send_msg` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3744:26
     |
3744 |         msg.id = context.send_msg(chat_id, &mut msg).await?;
     |                          ^^^^^^^^ help: there is a method with a similar name: `send_sync_msg`

error[E0061]: this method takes 1 argument but 3 arguments were supplied
    --> src/chat.rs:3746:13
     |
3746 |     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |             ^^^^^^^^^^                         -------  - unexpected argument of type `{integer}`
     |                                                |
     |                                                unexpected argument of type `ChatId`
     |
note: method defined here
    --> src/context.rs:595:12
     |
595  |     pub fn emit_event(&self, event: EventType) {
     |            ^^^^^^^^^^        ----------------
help: remove the extra arguments
     |
3746 -     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
3746 +     context.emit_event(DC_EVENT_CHAT_MODIFIED).await;
     |

error[E0277]: `()` is not a future
    --> src/chat.rs:3746:60
     |
3746 |     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |                                                           -^^^^^
     |                                                           ||
     |                                                           |`()` is not a future
     |                                                           help: remove the `.await`
     |
     = help: the trait `futures::Future` is not implemented for `()`, which is required by `(): std::future::IntoFuture`
     = note: () must be a future or must implement `IntoFuture` to be awaited
     = note: required for `()` to implement `std::future::IntoFuture`

error[E0599]: no method named `load_from_db` found for struct `chat::Chat` in the current scope
    --> src/chat.rs:3701:17
     |
1480 | pub struct Chat {
     | --------------- method `load_from_db` not found for this struct
...
3701 |         || chat.load_from_db(context).await? == 0 {
     |            -----^^^^^^^^^^^^---------
     |            |    |
     |            |    this is an associated function, not a method
     |            help: use associated function syntax instead: `chat::Chat::load_from_db(context)`
     |
     = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
note: the candidate is defined in an impl for the type `chat::Chat`
    --> src/chat.rs:1515:5
     |
1515 |     pub async fn load_from_db(context: &Context, chat_id: ChatId) -> Result<Self> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `get_config` found for struct `sql::Sql` in the current scope
    --> src/chat.rs:3715:34
     |
3715 |     self_addr = Some(context.sql.get_config("configured_addr", "").to_string());
     |                                  ^^^^^^^^^^ help: there is a method with a similar name: `get_raw_config`
     |
    ::: src/sql.rs:57:1
     |
57   | pub struct Sql {
     | -------------- method `get_config` not found for this struct

error[E0616]: field `addr` of struct `contact::Contact` is private
    --> src/chat.rs:3716:16
     |
3716 |     if contact.addr.eq_ignore_ascii_case(&self_addr.unwrap()) {
     |                ^^^^ private field

error[E0616]: field `addr` of struct `contact::Contact` is private
    --> src/chat.rs:3740:78
     |
3740 |         msg.text = dc_stock_system_msg(context, DC_STR_MSGADDMEMBER, contact.addr.as_str(), None, ContactId::SELF).await;
     |                                                                              ^^^^ private field

error[E0616]: field `addr` of struct `contact::Contact` is private
    --> src/chat.rs:3742:49
     |
3742 |         msg.param.set(DC_PARAM_CMD_ARG, contact.addr.as_str());
     |                                                 ^^^^ private field

error[E0061]: this method takes 1 argument but 3 arguments were supplied
    --> src/chat.rs:3746:13
     |
3746 |     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
     |             ^^^^^^^^^^                         -------  - unexpected argument of type `{integer}`
     |                                                |
     |                                                unexpected argument of type `chat::ChatId`
     |
note: method defined here
    --> src/context.rs:595:12
     |
595  |     pub fn emit_event(&self, event: EventType) {
     |            ^^^^^^^^^^        ----------------
help: remove the extra arguments
     |
3746 -     context.emit_event(DC_EVENT_CHAT_MODIFIED, chat_id, 0).await;
3746 +     context.emit_event(DC_EVENT_CHAT_MODIFIED).await;
     |

Some errors have detailed explanations: E0061, E0277, E0308, E0425, E0599, E0609, E0616.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `deltachat` (lib) due to 48 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `deltachat` (lib test) due to 48 previous errors
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

